agent/04_agent_prepare_release.sh

```diff
@@ -0,0 +1,96 @@
+#!/usr/bin/env bash
+set -euxo pipefail
+
+SCRIPTDIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )/.." && pwd )"
+
+LOGDIR=${SCRIPTDIR}/logs
+source $SCRIPTDIR/logging.sh
+source $SCRIPTDIR/common.sh
+source $SCRIPTDIR/network.sh
+source $SCRIPTDIR/utils.sh
+source $SCRIPTDIR/validation.sh
+source $SCRIPTDIR/agent/common.sh
+source $SCRIPTDIR/ocp_install_env.sh
+
+# To replace an image entry in the openshift release image, set <ENTRYNAME>_LOCAL_REPO so that:
+# - ENTRYNAME matches an uppercase version of the name in the release image with "-" converted to "_" 
+# - The var value must point to an already locally cloned repo
+#
+# To specify a custom Dockerfile set <ENTRYNAME>_DOCKERFILE, as a relative path of the Dockerfile
+# within the configured repo
+#
+# For example, to use a custom installer image:
+# export INSTALLER_LOCAL_REPO=~/go/src/github.com/openshift/installer
+# export INSTALLER_DOCKERFILE=images/installer/Dockerfile.ci
+
+early_deploy_validation
+write_pull_secret
+
+# Release mirroring could be required by the subsequent steps
+# even if the current one will be skipped
+if [[ ! -z "${MIRROR_IMAGES}" ]]; then
+   setup_release_mirror
+fi
+
+function build_local_release() {
+    # Sanity checks
+    if [ -z "${MIRROR_IMAGES}" ] ; then 
+        echo "Please set MIRROR_IMAGES to rebuild a local release"
+        exit 1
+    fi
+
+    if [[ ! "$OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE" =~ "$LOCAL_REGISTRY_DNS_NAME" ]] ; then
```

Comment:
I wonder why this check is necessary. Doesn't it conflict with this setting https://github.com/openshift-metal3/dev-scripts/blob/master/common.sh#L206 ?

Reply:
It should not. It's a sanity check to ensure that the subsequent [push operation](https://github.com/openshift-metal3/dev-scripts/pull/1472/files/e1b28eea8bb8be3d1aee47602d39c07c8daafee3#diff-a53878775eb60ffffce541ffc1a8f37dcb1aeb39b6526e93b733a33fd510c0b4R87) will point to the local registry